airflow/providers/amazon/aws/operators/redshift_to_s3.py

```diff
@@ -107,7 +112,12 @@ def execute(self, context):
         credentials = s3_hook.get_credentials()
         unload_options = '\n\t\t\t'.join(self.unload_options)
         s3_key = '{}/{}_'.format(self.s3_key, self.table) if self.table_as_file_name else self.s3_key
-        select_query = "SELECT * FROM {schema}.{table}".format(schema=self.schema, table=self.table)
+        if all([self.schema, self.table]):
+            select_query = "SELECT * FROM {schema}.{table}".format(schema=self.schema, table=self.table)
+        elif self.custom_select_query:
+            select_query = self.custom_select_query
+        else:
+            raise AirflowBadRequest(f"Either (schema, table) or custom_select_query should be set. They are ({self.schema},{self.table}) and {self.custom_select_query} now.")
```

Comment:
This validation should be made during operator `__init__` (or at least before initiating the hooks).

Reply:
Makes sense. I will make the change. Thanks!